DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,195 @@
+
+            var duckResultFactory = DuckResultFactory.Create(apiResult, duckResult.SearchedText);
+
+            DuckResult result = duckResult.SearchResultType switch
+            {
+                ResultType.Answer => duckResult, // Answers will often have numerical computations and are not suitable for custom tags as they often update very frequently.
+                ResultType.Definition => duckResultFactory.GetDefinition(),
+                ResultType.Abstract => duckResultFactory.GetAbstract(),
+                ResultType.QrCode => null,
+                ResultType.SearchResult => duckResult,
+                _ => null
+            };
+
+            return result == null ? default(ISearchResult) : GetISearchResult(result);
+        }
+
+        public ValueTask<IHandleResult> HandleSearchResult(ISearchResult searchResult)
+        {
+            if (searchResult is not DuckDuckGoSearchResult duckGoSearchResult)
+                throw new InvalidCastException(nameof(DuckDuckGoSearchResult));
+
+            string url = duckGoSearchResult.Url;
+
+            if (duckGoSearchResult.SelectedOperation is DuckDuckGoSearchOperations duckGoSearchOperations)
+            {
+                IProcessManager managerInstance = ProcessUtils.GetManagerInstance();
+                switch (duckGoSearchOperations.ActionType)
+                {
+                    case ActionType.OpenDuckDuckGo:
+                        if (!string.IsNullOrWhiteSpace(url))
+                            managerInstance.StartNewProcess(url);
+                        break;
+
+                    case ActionType.SaveImage:
+                        SaveQrImage(duckGoSearchResult);
+                        break;
+
+                    default:
+                        return default;
+                }
+            }
+            else if (duckGoSearchResult.SelectedOperation.OperationName == CopyUrl.OperationName)
+            {
+                if (!string.IsNullOrWhiteSpace(url))
+                    Clipboard.SetText(url);
+            }
+            else if (duckGoSearchResult.SelectedOperation.OperationName == CopyContents.OperationName)
+            {
+                if (!string.IsNullOrWhiteSpace(duckGoSearchResult.ResultName))
+                    Clipboard.SetText(duckGoSearchResult.ResultName);
+            }
+
+            return new ValueTask<IHandleResult>(new HandleResult(true, false));
+        }
+
+        private DuckDuckGoSearchResult GetISearchResult(DuckResult duckResult)
+        {
+            double score = duckResult.Info.SearchDistanceScore(duckResult.SearchedText);
+            return new DuckDuckGoSearchResult(duckResult.Info, duckResult.SearchedText, duckResult.ResultType,
+                DuckOperations, score, duckResult)
+            {
+                Url = duckResult.SourceUrl, AdditionalInformation = duckResult.SourceUrl,
+                PreviewImage = _logoImage, IsPinned = duckResult.IsPinned
```

Comment:
Why do you set `IsPinned`? It should be set by the user

Reply:
It was used to give the highest possible priority to a search result. Instant answers should always be on top like in Search Engines. Now I've removed it and used `score` instead.